version: "3.4"
services:
  cache:
    ports:
      - "11211:11211"
    networks:
      - ebau.local

  proxy:
    volumes:
      - ../proxy/kt_gr.conf:/etc/nginx/conf.d/default.conf:ro
    ports:
      - "80:80"
    networks:
      ebau.local:
        aliases:
          - ember-ebau.local
          - ebau-portal.local
          - ebau-rest-portal.local
          - ebau-keycloak.local

  db:
    # Workaround, same name of image as cache_from
    # better be named differently once following is fixed:
    # https://github.com/docker/compose/issues/5458
    # see also other services below
    image: acr.run/camac-ng/camac-ng/db:master
    ports:
      - "5432:5432"
    environment:
      - POSTGRES_PASSWORD=camac
    volumes:
      - ../db/init-user-db.sh:/docker-entrypoint-initdb.d/init-user-db.sh:ro
    networks:
      - ebau.local

  django: &django
    image: acr.run/camac-ng/camac-ng/django:master
    build:
      args:
        REQUIREMENTS: requirements-dev.txt
    ports:
      - "5555:5555"
      - "5678:5678"
    volumes:
      - ../django:/app
    environment:
      - ENV=development
      - APPLICATION_ENV=development
      - KEYCLOAK_URL=http://ebau-keycloak.local/auth/
      - DATABASE_PASSWORD=camac
      - DATABASE_USER=camac
      - DJANGO_EMAIL_HOST=mailhog
      - DJANGO_EMAIL_PORT=1025
      - OAUTHLIB_INSECURE_TRANSPORT=1
      - OIDC_USERINFO_ENDPOINT=http://ebau-keycloak.local/auth/realms/ebau/protocol/openid-connect/userinfo
      - PYTHONDONTWRITEBYTECODE=1
      - GWR_FERNET_KEY=GUdwT6yKxqfszrnHC7wyKDmtVZdhjSVQEPnpeJYNLf8=
    networks:
      - ebau.local
    command: /bin/sh -c "./wait-for-it.sh db:5432 -- ./manage.py migrate && ./manage.py collectstatic --noinput && ./manage.py runserver 0:80 --pythonpath /app/$APPLICATION"
    # command: /bin/sh -c "while true; do sleep 10; done"

  ember-caluma-portal:
    image: acr.run/camac-ng/camac-ng/ember-caluma-portal:master
    networks:
      - ebau.local

  ember-ebau:
    image: acr.run/camac-ng/camac-ng/ember-ebau:master
    networks:
      - ebau.local

  keycloak:
    image: acr.run/camac-ng/camac-ng/keycloak-${APPLICATION}:master
    volumes:
      - ../keycloak/config/${APPLICATION}-test-config.json:/opt/keycloak/data/import/test-config.json
    environment:
      - KC_DB_PASSWORD=camac
      - KC_DB_URL=jdbc:postgresql://db:5432/${APPLICATION}?currentSchema=keycloak
      - KC_DB_USERNAME=camac
      - KC_HOSTNAME=ebau-keycloak.local
      - KEYCLOAK_ADMIN_PASSWORD=camac
    networks:
      - ebau.local

  mailhog:
    networks:
      - ebau.local
    environment:
      - MH_UI_WEB_PATH=mailhog

  document-merge-service:
    networks:
      - ebau.local
    environment:
      - ENV=dev
      - OIDC_USERINFO_ENDPOINT=http://ebau-keycloak.local/auth/realms/ebau/protocol/openid-connect/userinfo
      - OIDC_GROUPS_API=http://ebau.local/api/v1/me?include=service,service.service_parent
    # There appears to be bug where libreoffice can only be started as the root
    # user. The workaround of fixing owner and permissions of ~/.config did not
    # work in our case.
    #
    # https://github.com/unoconv/unoconv/issues/241#issuecomment-549434601
    # https://superuser.com/questions/564534/only-root-user-is-able-to-open-libreoffice
    # user: "${UID:?Set UID env variable to your user id}"
    command:
      [
        "/bin/sh",
        "-c",
        "wait-for-it.sh db:5432 -- poetry run python manage.py migrate && poetry run uwsgi",
      ]

  alexandria:
    volumes:
      - ../alexandria/kt_gr:/app/alexandria/kt_gr
    ports:
      - "8000:8000"
    depends_on:
      - db
      - minio
    environment:
      - DATABASE_PASSWORD=camac
      - ALLOWED_HOSTS=*
      - OIDC_OP_USER_ENDPOINT=http://ebau-keycloak.local/auth/realms/ebau/protocol/openid-connect/userinfo
      - MINIO_STORAGE_ENDPOINT=ember-ebau.local
      - MINIO_STORAGE_SECRET_KEY=minio123
    networks:
      - ebau.local

  minio:
    environment:
      - MINIO_SECRET_KEY=minio123
    networks:
      - ebau.local

  mc:
    networks:
      - ebau.local

networks:
  ebau.local:
